plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,538 @@
+    type: str
+  description:
+    description:
+    - The description of the L3Out Interface Group Policy.
+    type: str
+  name:
+    description:
+    - The name of the L3Out Interface Group Policy.
+    type: str
+    aliases: [ l3out_interface_group_policy ]
+  interface_routing_policy:
+    description:
+    - The configuration of the Interface Routing Policy.
+    - Providing an empty dictionary will remove the O(interface_routing_policy={}) from L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the L3Out Interface Routing Policy.
+        type: str
+      template:
+        description:
+        - The template associated with the L3Out Interface Routing Policy.
+        type: str
+  interface_routing_policy_uuid:
+    description:
+    - The UUID of the L3Out Interface Routing Policy.
+    - Providing an empty string will remove the O(interface_routing_policy_uuid="") from L3Out Interface Group Policy.
+    type: str
+  bfd:
+    description:
+    - The BFD configuration.
+    - Providing an empty dictionary will remove the O(bfd={}) from the L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      enable:
+        description:
+        - Whether BFD is enabled.
+        type: bool
+      authentication:
+        description:
+        - Whether BFD authentication is enabled.
+        type: bool
+      key_id:
+        description:
+        - The BFD key ID.
+        type: int
+      key:
+        description:
+        - The BFD key.
+        type: str
+  bfd_multi_hop:
+    description:
+    - The BFD multi-hop configuration.
+    - Providing an empty dictionary will remove the O(bfd_multi_hop={}) from the L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      enable:
+        description:
+        - Whether BFD multi-hop is enabled.
+        type: bool
+      authentication:
+        description:
+        - Whether BFD multi-hop authentication is enabled.
+        type: bool
+      key_id:
+        description:
+        - The BFD multi-hop key ID.
+        type: int
+      key:
+        description:
+        - The BFD multi-hop key.
+        type: str
+  ospf:
+    description:
+    - The OSPF configuration.
+    - Providing an empty dictionary will remove the O(ospf={}) from the L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      enable:
+        description:
+        - Whether OSPF is enabled.
+        type: bool
+      authentication_type:
+        description:
+        - The type of OSPF authentication.
+        type: str
+        choices: [ simple, md5, none ]
+      key_id:
+        description:
+        - The OSPF key ID.
+        type: int
+      key:
+        description:
+        - The OSPF key.
+        type: str
+  qos_priority:
+    description:
+    - The QoS priority level.
+    type: str
+    choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+  custom_qos_policy:
```

Comment:
The `Custom QoS Policy` is unique within the tenant level, so only `Custom QoS Policy` name is enough. Please recheck the logic once again.